### PR TITLE
Set TransactionGasCalculator in ETC specs

### LINKED ISSUE #2715: DOCKER: Invalid block: receipts root mismatch @v21.7.3
### Steps to Reproduce (Bug)
1. Have a running working v21.7.2 version in docker
2. Upgrade to v21.7.3

**Expected behavior:**
```besu_1           | 2021-09-05 15:52:59.458+00:00 | nioEventLoopGroup-3-10 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:53:04.298+00:00 | nioEventLoopGroup-3-9 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:53:09.302+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:53:14.304+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:53:19.305+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:53:19.529+00:00 | nioEventLoopGroup-3-5 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:53:24.061+00:00 | nioEventLoopGroup-3-1 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 3
besu_1           | 2021-09-05 15:53:29.062+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:53:34.305+00:00 | nioEventLoopGroup-3-10 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0xce7cbef463c5bac7e3... at block 13470991
besu_1           | 2021-09-05 15:53:48.932+00:00 | EthScheduler-Services-5 (importBlock) | INFO  | FullImportBlockStep | Import reached block 13471000 (0x7694d01aa8e2ad5cff6e4f30dd882782e7f23f5e346066dab4a26108ccfaaf46), - Mg/s, Peers: 1
besu_1           | 2021-09-05 15:53:56.482+00:00 | EthScheduler-Services-5 (importBlock) | INFO  | FullImportBlockStep | Import reached block 13471200 (0x8a32ed0c9debb222e95f43e72bdd31266c010d68a77c7ac639f8d1cfd4fb6fbb), 4.260 Mg/s, Peers: 1
besu_1           | 2021-09-05 15:54:04.692+00:00 | EthScheduler-Services-5 (importBlock) | INFO  | FullImportBlockStep | Import reached block 13471400 (0x64d03b0d0b6b63de54d7031fbf4fb511fdbcf4c6442b5d14db77aa2224d6418a), 6.134 Mg/s, Peers: 1
besu_1           | 2021-09-05 15:54:09.825+00:00 | EthScheduler-Services-5 (importBlock) | INFO  | FullImportBlockStep | Import reached block 13471600 (0xe301e740821d93ff3541d64b0c4a40628bf289610ac2db5f083d697d555d5d30), 5.073 Mg/s, Peers: 2
besu_1           | 2021-09-05 15:54:14.820+00:00 | EthScheduler-Services-5 (importBlock) | INFO  | FullImportBlockStep | Import reached block 13471800 (0x0fee8e1b95da9ce306c496659a3b8e695568ebced0ff45ce6430fd80d0243e1d), 6.821 Mg/s, Peers: 1
besu_1           | 2021-09-05 15:54:18.125+00:00 | nioEventLoopGroup-3-10 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0xce7cbef463c5bac7e3... at block 13471861
besu_1           | 2021-09-05 15:54:35.240+00:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #13,471,866 / 9 tx / 0 om / 189,000 (1.9%) gas / (0x816bf44378d63bfd0f73a1327095f7789a35b6cb8239d030fb636e63a28e315c) in 0.027s. Peers: 1
besu_1           | 2021-09-05 15:54:36.928+00:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #13,471,867 / 0 tx / 0 om / 0 (0.0%) gas / (0x6403bdf63cc05189c587cf1ceeb47d47d2fdb5b443acdb8283c4022c8ba84100) in 0.002s. Peers: 2
besu_1           | 2021-09-05 15:55:47.821+00:00 | EthScheduler-Workers-2 | INFO  | PersistBlockTask | Imported #13,471,868 / 22 tx / 0 om / 521,105 (5.2%) gas / (0x84a8a8b67e80453c1d02a55970d1751120cae7eb6d783160fb5bc0777b223053) in 0.094s. Peers: 1
besu_1           | 2021-09-05 15:56:03.608+00:00 | EthScheduler-Workers-0 | INFO  | PersistBlockTask | Imported #13,471,869 / 4 tx / 0 om / 85,012 (0.9%) gas / (0xf750ebc80b973a300aa2c28483814784bb8f4b75c379273b6b6da777ddb8366a) in 0.024s. Peers: 1
besu_1           | 2021-09-05 15:56:19.157+00:00 | EthScheduler-Workers-2 | INFO  | PersistBlockTask | Imported #13,471,870 / 8 tx / 0 om / 168,000 (1.7%) gas / (0x7b29db5f5af7077116585f4cce8a6795689e70432f73b20c3b0ebea4e50f268e) in 0.033s. Peers: 1
besu_1           | 2021-09-05 15:56:21.369+00:00 | EthScheduler-Workers-3 | INFO  | PersistBlockTask | Imported #13,471,871 / 0 tx / 0 om / 0 (0.0%) gas / (0x429dabc418db6c41659d432d9b56eb0fceda52bbcb8c0ed9302ef75bf9bb46ec) in 0.003s. Peers: 1```
````

**Actual behavior:**

```besu_1           | 2021-09-05 15:48:50.636+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:48:55.649+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:00.661+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:05.663+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:10.667+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:15.668+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:49:20.477+00:00 | nioEventLoopGroup-3-6 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:25.478+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:49:30.481+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:49:35.483+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:49:40.484+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:49:45.486+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:49:45.795+00:00 | nioEventLoopGroup-3-6 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:50.796+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:49:55.797+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:50:00.799+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:50:05.801+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:50:10.803+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:50:15.847+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:50:21.420+00:00 | nioEventLoopGroup-3-6 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0x2acbfae44f20064f9c... at block 13470991
besu_1           | 2021-09-05 15:50:23.999+00:00 | EthScheduler-Services-11 (importBlock) | WARN  | MainnetBlockBodyValidator | Invalid block: receipts root mismatch (expected=0x7b9abd93ad0004e42a3293e7c84d02c8d034a5013b53297b724d8ba755cda188, actual=0x53022f47a30c7bf359b3a3c309a233376959d35943db23f346dcc7a83fb0e4d1)
besu_1           | 2021-09-05 15:50:24.003+00:00 | EthScheduler-Services-11 (importBlock) | WARN  | PipelineChainDownloader | Invalid block detected.  Disconnecting from sync target. Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           | 2021-09-05 15:50:24.012+00:00 | EthScheduler-Services-11 (importBlock) | WARN  | PipelineChainDownloader | Chain download failed. Restarting after short delay.
besu_1           | java.util.concurrent.CompletionException: org.hyperledger.besu.ethereum.eth.sync.tasks.exceptions.InvalidBlockException: Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.completeRelay(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture$UniRelay.tryFire(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.abort(Pipeline.java:180)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.lambda$runWithErrorHandling$3(Pipeline.java:159)
besu_1           |      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
besu_1           |      at java.base/java.lang.Thread.run(Unknown Source)
besu_1           | Caused by: org.hyperledger.besu.ethereum.eth.sync.tasks.exceptions.InvalidBlockException: Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           |      at org.hyperledger.besu.ethereum.eth.sync.fullsync.FullImportBlockStep.accept(FullImportBlockStep.java:55)
besu_1           |      at org.hyperledger.besu.ethereum.eth.sync.fullsync.FullImportBlockStep.accept(FullImportBlockStep.java:31)
besu_1           |      at org.hyperledger.besu.services.pipeline.CompleterStage.run(CompleterStage.java:37)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.lambda$runWithErrorHandling$3(Pipeline.java:152)
besu_1           |      ... 5 more
besu_1           | 2021-09-05 15:50:26.018+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:50:29.101+00:00 | nioEventLoopGroup-3-9 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:50:34.105+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:50:34.293+00:00 | Thread-1 | INFO  | DefaultSynchronizer | Stopping synchronizer
besu_1           | 2021-09-05 15:50:34.295+00:00 | Thread-1 | INFO  | NetworkRunner | Stopping Network.
besu_1           | 2021-09-05 15:50:34.325+00:00 | Thread-1 | INFO  | EthProtocolManager | Stopping eth Subprotocol.
besu_1           | 2021-09-05 15:50:34.425+00:00 | Thread-1 | INFO  | EthProtocolManager | eth Subprotocol stopped.
besu_1           | 2021-09-05 15:50:34.426+00:00 | Thread-1 | INFO  | NetworkRunner | Network stopped.
besu_1           | 2021-09-05 15:50:34.426+00:00 | Thread-1 | INFO  | AutoTransactionLogBloomCachingService | Shutting down Auto transaction logs caching service.
besu_1           | 2021-09-05 15:50:34.433+00:00 | Thread-1 | INFO  | DockerNatManager | Stopping docker NAT manager.
besu_1           | 2021-09-05 15:51:07.575+00:00 | main | INFO  | Besu | Using LibEthPairings native alt bn128
besu_1           | 2021-09-05 15:51:07.582+00:00 | main | INFO  | Besu | Using the native implementation of the signature algorithm
besu_1           | 2021-09-05 15:51:07.593+00:00 | main | INFO  | Besu | Starting Besu version: besu/v21.7.3/linux-x86_64/adoptopenjdk-java-11
besu_1           | 2021-09-05 15:51:08.195+00:00 | main | INFO  | Besu | Static Nodes file = /var/lib/besu/static-nodes.json
besu_1           | 2021-09-05 15:51:08.208+00:00 | main | INFO  | StaticNodesParser | StaticNodes file /var/lib/besu/static-nodes.json does not exist, no static connections will be created.
besu_1           | 2021-09-05 15:51:08.209+00:00 | main | INFO  | Besu | Connecting to 0 static nodes.
besu_1           | 2021-09-05 15:51:08.217+00:00 | main | INFO  | Besu | Security Module: localfile
besu_1           | 2021-09-05 15:51:08.255+00:00 | main | INFO  | DatabaseMetadata | Lookup database metadata file in data directory: /var/lib/besu
besu_1           | 2021-09-05 15:51:08.481+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | Existing database detected at /var/lib/besu. Version 1
besu_1           | 2021-09-05 15:51:10.548+00:00 | main | INFO  | KeyPairUtil | Loaded public key 0xc029faa1d8eb410685f1eb0b9bc53e9e330ecffae9c40effd9b6297964a22c579a79c06e1b9ce3c0d57808c295fd34a5675c94d525dc46e8d0a4f78e7ac671ed from /var/lib/besu/key
besu_1           | 2021-09-05 15:51:11.108+00:00 | main | INFO  | ProtocolScheduleBuilder | Protocol schedule created with milestones: [Frontier: 0, Homestead: 1150000, ClassicRecoveryInit: 1920000, Homestead: 1920001, TangerineWhistle: 2500000, DieHard: 3000000, Gotham: 5000000, DefuseDifficultyBomb: 5900000, Atlantis: 8772000, Agharta: 9573000, Phoenix: 10500839, Thanos: 11700000, Magneto: 13189133]
besu_1           | 2021-09-05 15:51:12.768+00:00 | main | INFO  | FastDownloaderFactory | Fast sync was requested, but cannot be enabled because the local blockchain is not empty.
besu_1           | 2021-09-05 15:51:12.803+00:00 | main | INFO  | RunnerBuilder | Detecting NAT service.
besu_1           | 2021-09-05 15:51:13.296+00:00 | main | INFO  | Runner | Starting Ethereum main loop ...
besu_1           | 2021-09-05 15:51:13.298+00:00 | main | INFO  | MetricsHttpService | Starting metrics http service on 0.0.0.0:9545
besu_1           | 2021-09-05 15:51:13.652+00:00 | main | INFO  | DockerNatManager | Starting docker NAT manager.
besu_1           | 2021-09-05 15:51:13.652+00:00 | vert.x-eventloop-thread-2 | INFO  | MetricsHttpService | Metrics service started and listening on 9545:9545
besu_1           | 2021-09-05 15:51:13.743+00:00 | main | INFO  | NetworkRunner | Starting Network.
besu_1           | 2021-09-05 15:51:13.797+00:00 | main | INFO  | RlpxAgent | P2P RLPx agent started and listening on /0.0.0.0:30303.
besu_1           | 2021-09-05 15:51:13.799+00:00 | main | INFO  | PeerDiscoveryAgent | Starting peer discovery agent on host=0.0.0.0, port=30303
besu_1           | 2021-09-05 15:51:13.877+00:00 | vert.x-eventloop-thread-3 | INFO  | VertxPeerDiscoveryAgent | Started peer discovery agent successfully, on effective host=0.0.0.0 and port=30303
besu_1           | 2021-09-05 15:51:13.882+00:00 | vert.x-eventloop-thread-3 | INFO  | PeerDiscoveryAgent | P2P peer discovery agent started and listening on /0.0.0.0:30303
besu_1           | 2021-09-05 15:51:14.033+00:00 | vert.x-eventloop-thread-3 | INFO  | PeerDiscoveryAgent | Writing node record to disk. NodeRecord{seq=70, publicKey=0x03c029faa1d8eb410685f1eb0b9bc53e9e330ecffae9c40effd9b6297964a22c57, udpAddress=Optional[/127.0.0.1:30303], tcpAddress=Optional[/127.0.0.1:30303], asBase64=-Je4QEsqku572BXquNXUkhN-G05ikmUGbQnz3iZfCi_dy9-KKAs1IGmEoW7hUL28FsLkprJ1JjR0H808fUcyfhnQ_c5Gg2V0aMfGhA9r8YeAgmlkgnY0gmlwhH8AAAGJc2VjcDI1NmsxoQPAKfqh2OtBBoXx6wubxT6eMw7P-unEDv_Ztil5ZKIsV4N0Y3CCdl-DdWRwgnZf, nodeId=0xee4f90949522ca8ddf6dbe7e2cbb13b80b2908d35bd245f62285af040709f82b, customFields={tcp=30303, udp=30303, ip=0x7f000001, eth=[[0x0f6bf187, 0x]], id=V4, secp256k1=0x03c029faa1d8eb410685f1eb0b9bc53e9e330ecffae9c40effd9b6297964a22c57}}
besu_1           | SLF4J: Class path contains multiple SLF4J bindings.
besu_1           | SLF4J: Found binding in [jar:file:/opt/besu/lib/log4j-slf4j-impl-2.14.1.jar!/org/slf4j/impl/StaticLoggerBinder.class]
besu_1           | SLF4J: Found binding in [jar:file:/opt/besu/lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
besu_1           | SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
besu_1           | SLF4J: Actual binding is of type [org.apache.logging.slf4j.Log4jLoggerFactory]
besu_1           | 2021-09-05 15:51:14.326+00:00 | main | INFO  | DefaultP2PNetwork | Enode URL enode://c029faa1d8eb410685f1eb0b9bc53e9e330ecffae9c40effd9b6297964a22c579a79c06e1b9ce3c0d57808c295fd34a5675c94d525dc46e8d0a4f78e7ac671ed@127.0.0.1:30303
besu_1           | 2021-09-05 15:51:14.329+00:00 | main | INFO  | DefaultP2PNetwork | Node address 0x2cbb13b80b2908d35bd245f62285af040709f82b
besu_1           | 2021-09-05 15:51:14.338+00:00 | main | INFO  | DefaultSynchronizer | Starting synchronizer.
besu_1           | 2021-09-05 15:51:14.373+00:00 | main | INFO  | FullSyncDownloader | Starting full sync.
besu_1           | 2021-09-05 15:51:14.388+00:00 | main | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:51:14.436+00:00 | main | INFO  | JsonRpcHttpService | Starting JSON-RPC service on 0.0.0.0:8545
besu_1           | 2021-09-05 15:51:14.507+00:00 | vert.x-eventloop-thread-7 | INFO  | JsonRpcHttpService | JSON-RPC service started and listening on 0.0.0.0:8545
besu_1           | 2021-09-05 15:51:14.515+00:00 | main | INFO  | Runner | Ethereum main loop is up.
besu_1           | 2021-09-05 15:51:14.525+00:00 | main | INFO  | AutoTransactionLogBloomCachingService | Starting auto transaction log bloom caching service.
besu_1           | 2021-09-05 15:51:14.532+00:00 | main | INFO  | LogBloomCacheMetadata | Lookup cache metadata file in data directory: /var/lib/besu/caches
besu_1           | 2021-09-05 15:51:16.242+00:00 | nioEventLoopGroup-3-3 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:51:21.246+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:51:21.830+00:00 | nioEventLoopGroup-3-7 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:51:26.831+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:51:31.833+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:51:36.835+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
besu_1           | 2021-09-05 15:51:41.839+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:51:44.547+00:00 | nioEventLoopGroup-3-3 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:51:48.720+00:00 | nioEventLoopGroup-3-8 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:51:53.724+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:51:58.725+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:52:03.726+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:52:06.747+00:00 | nioEventLoopGroup-3-6 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 2
besu_1           | 2021-09-05 15:52:11.753+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
besu_1           | 2021-09-05 15:52:14.868+00:00 | nioEventLoopGroup-3-7 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 3
besu_1           | 2021-09-05 15:52:20.036+00:00 | nioEventLoopGroup-3-8 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0x1abea5c06b5f22e9f0... at block 13470991
besu_1           | 2021-09-05 15:52:33.755+00:00 | EthScheduler-Services-5 (importBlock) | WARN  | MainnetBlockBodyValidator | Invalid block: receipts root mismatch (expected=0x7b9abd93ad0004e42a3293e7c84d02c8d034a5013b53297b724d8ba755cda188, actual=0x53022f47a30c7bf359b3a3c309a233376959d35943db23f346dcc7a83fb0e4d1)
besu_1           | 2021-09-05 15:52:33.819+00:00 | EthScheduler-Services-5 (importBlock) | WARN  | PipelineChainDownloader | Invalid block detected.  Disconnecting from sync target. Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           | 2021-09-05 15:52:33.830+00:00 | EthScheduler-Services-5 (importBlock) | WARN  | PipelineChainDownloader | Chain download failed. Restarting after short delay.
besu_1           | java.util.concurrent.CompletionException: org.hyperledger.besu.ethereum.eth.sync.tasks.exceptions.InvalidBlockException: Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.encodeRelay(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.completeRelay(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture$UniRelay.tryFire(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.abort(Pipeline.java:180)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.lambda$runWithErrorHandling$3(Pipeline.java:159)
besu_1           |      at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
besu_1           |      at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
besu_1           |      at java.base/java.lang.Thread.run(Unknown Source)
besu_1           | Caused by: org.hyperledger.besu.ethereum.eth.sync.tasks.exceptions.InvalidBlockException: Failed to import block: Invalid block at #13470992 (0x7ddba3f3ea2bee8aa42088d370326fdbc017be3fa2448a4070eb5d7b50b261ba)
besu_1           |      at org.hyperledger.besu.ethereum.eth.sync.fullsync.FullImportBlockStep.accept(FullImportBlockStep.java:55)
besu_1           |      at org.hyperledger.besu.ethereum.eth.sync.fullsync.FullImportBlockStep.accept(FullImportBlockStep.java:31)
besu_1           |      at org.hyperledger.besu.services.pipeline.CompleterStage.run(CompleterStage.java:37)
besu_1           |      at org.hyperledger.besu.services.pipeline.Pipeline.lambda$runWithErrorHandling$3(Pipeline.java:152)
besu_1           |      ... 5 more
```
**Frequency:** 100%

### Versions
* Docker Version: Docker Latest (v21.7.3)

### PR DESCRIPTION
**Pull request by diega**
## PR description
On #2659 the gas calculator was split but the ETC specification wasn't properly updated, letting 21.7.3 nodes out of consensus.

## Fixed Issue(s)
Fixes #2715

## Changelog

- [ ] Update Ethereum Classic Phoenix and Magneto specs

### CODE REVIEW

**Review by atoulme — APPROVED** [review]
LGTM let’s see what your manual testing does

**Comment by macfarla** [comment]
@atoulme @diega is this waiting on manual testing mentioned above?

**Comment by diega** [comment]
> @atoulme @diega is this waiting on manual testing mentioned above?

Yes, I'm syncing BTC mainnet from scratch. After 5 days I'm at block ~9M.
These changes would affect __Phoenix__ at block 10500839 and __Magneto__ at block 13189133. Previous to this big end-to-end testing I had a `21.1.7` up to date node which lost peers once I upgraded to `21.7.3` b/c of headers mismatch. Once I applied the patch, it continued the sync just fine.

**Comment by atoulme** [comment]
@macfarla I'll let @diega merge his changes when he feels good about them, I approved them as they make sense to me.

**Comment by macfarla** [comment]
> @macfarla I'll let @diega merge his changes when he feels good about them, I approved them as they make sense to me.

@diega you'll need a maintainer to hit the merge button so shout out when you're happy with the testing!

**Comment by bobsummerwill** [comment]
Thanks, @atoulme and @macfarla!

**Comment by diega** [comment]
I just full synced the _Mordor_ testnet with this patch applied and everything went fine, so I can consider my manual testing a success. I think I'll need some maintainer to run the checks again b/c I just rebased my branch on top of `main`

**Pull request merged**
